libraries/stdlib/osl/stdosl.h

```diff
@@ -573,6 +573,221 @@ closure color volume_matte() BUILTIN;
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color translucent_bsdf(normal N, color albedo) BUILTIN;
+
+// Constructs a closure that represents straight transmission through a surface.
+//
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+// NOTE:
+//  - This is not a node in the MaterialX library, but the surface shader constructor
+//    node has an 'opacity' parameter to control textured cutout opacity.
+//
+closure color transparent_bsdf() BUILTIN;
+​
+// Constructs a BSSRDF for subsurface scattering within a homogeneous medium.
+//
+//  \param  N           Normal vector of the surface point beeing shaded.
+//  \param  albedo      Surface albedo.
+//  \param  sss_depth   Mean-free path in units of scene length.
+//  \param  sss_color   Scattering color / transmittance. The desired color resulting from white light transmitted a distance of 'sss_depth'
+//                      through the surface.
+//  \param  anisotropy  Scattering anisotropy [-1,1]. Negative values give backwards scattering, positive values give forward scattering, 
+//                      and 0.0 gives uniform scattering.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color subsurface_bssrdf(normal N, color albedo, float sss_depth, color sss_color, float anisotropy) BUILTIN;
+​
+// Constructs a microfacet BSDF for the back-scattering properties of cloth-like materials.
+// This closure may be vertically layered over a base BSDF, where energy that is not reflected
+// will be transmitted to the base closure.
+//
+//  \param  N           Normal vector of the surface point beeing shaded.
+//  \param  albedo      Surface albedo.
+//  \param  roughness   Surface roughness [0,1].
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color sheen_bsdf(normal N, color albedo, float roughness) BUILTIN;
+​
+​
+// -------------------------------------------------------------//
+// EDF closures                                                 //
+// -------------------------------------------------------------//
+​
+// Constructs an EDF emitting light uniformly in all directions.
+//
+//  \param  emittance   Radiant emittance of light leaving the surface.
+//  \param  label       Optional string parameter to name this component. For use in AOVs / LPEs.
+//
+closure color uniform_edf(color emittance) BUILTIN;
+​
+​
+// -------------------------------------------------------------//
+// VDF closures                                                 //
+// -------------------------------------------------------------//
+​
+// Constructs a VDF scattering light for a participating medium, based on the Henyey-Greenstein
+// phase function. Forward, backward and uniform scattering is supported and controlled by the
+// anisotropy input.
+//
```

Comment:
I'm not super keen on the term "VDF",  as what we are describing here is not a single function but really all the properties of the "medium" through which light is propagating (in the sense of the term "participating medium" in graphics, or [optical medium](https://en.wikipedia.org/wiki/Optical_medium) in physics).

But maybe it is a reasonable loose terminology (as one could consider all of albedo, extinction and anisotropy as representing the cross sections for scattering/absorption, which are generally functions of incident/output direction).

Reply:
Yep, I understand your point. My personal opinion is that it's better to have a simple and consistent naming of things (even if it's not 100% correct from a physical or mathematical point of view), then to have a complicated and inconsistent naming just to make sure the correct terminology from physics is used everywhere.

At least in the MaterialX library we are generalizing quite a bit, and consider a VDF to be "anything that has to do with light interacting with volumes / participating media", an EDF to be "anything that has to do with light emission" and a BSDF "anything that has to do with light interacting with surfaces". And there are some obvious incorrect uses of this, like for the node subsurface_bsdf which contains both surface and volume interactions, but we still call it a BSDF because it's used as a BSDF by artists. So an artist don't have to bother learning the difference between a BSSRDF and a BSDF.